happytransformer/happy_generation.py

```diff
@@ -0,0 +1,160 @@
+"""
+GEN_DEFAULT_SETTINGS = {
+    "do_sample": False,
+    "early_stopping": False,
+    "num_beams": 1,
+    "temperature": 1,
+    "top_k": 50,
+    "no_repeat_ngram_size": 0,
+}
+
+# greedy is prone to repetition loops. So, we'll set no_repeat_ngram_size to 2.
+GEN_GREEDY_SETTINGS = {
+    "do_sample": False,
+    "early_stopping": False,
+    "no_repeat_ngram_size": 2,
+}
+
+GEN_BEAM_SETTINGS = {
+    "do_sample": False,
+    "early_stopping": True,
+    "num_beams": 5,
+}
+
+GEN_GENERIC_SAMPLING_SETTINGS = {
+    "do_sample": True,
+    "early_stopping": False,
+    "top_k": 0,
+    "temperature": 0.7,
+}
+
+
+GEN_TOP_K_SAMPLING_SETTINGS = {
+    "do_sample": True,
+    "early_stopping": False,
+    "top_k": 50,
+    "temperature": 0.7,
+}
+
+
+@dataclass
+class GenerationResult:
+    text: str
+
+
+class HappyGeneration(HappyTransformer):
+    """
+    A user facing class for text generation
+    """
+    def __init__(self, model_type: str = "GPT2", model_name: str = "gpt2"):
+
+        self.adaptor = get_adaptor(model_type)
+
+        model = AutoModelForCausalLM.from_pretrained(model_name)
+
+        super().__init__(model_type, model_name, model)
+
+        self._trainer = GENTrainer(self.model, model_type, self.tokenizer, self._device, self.logger)
+
+    def __assert_default_text_is_val(self, text):
+
+        if not isinstance(text, str):
+            raise ValueError("The text input must be a string")
+        if not text:
+            raise ValueError("The text input must have at least one character")
+
+
+    def generate_text(self, text, settings=GEN_BEAM_SETTINGS,
+                      min_length=20, max_length=60) -> GenerationResult:
+        """
+        :param text: starting text that the model uses to generate text with.
+        :param settings: A dictionary that contains settings that determine what
+         algorithm is used to generate text
+        :param min_length: The minimum number of tokens for the output
+        :param max_length: The maximum number of tokens for the output
+        :return: Text that the model generates.
+        """
+
+        self.__assert_default_text_is_val(text)
+        self.__validate_settings(settings, min_length, max_length)
+
+        input_ids = self.tokenizer.encode(text, return_tensors="pt")
+        adjusted_min_length = min_length + len(input_ids[0])
+        adjusted_max_length = max_length + len(input_ids[0])
+
+        output = self.model.generate(input_ids,
+                                     **settings,
+                                     min_length=adjusted_min_length,
+                                     max_length=adjusted_max_length,
+                                     )
+        result = self.tokenizer.decode(output[0], skip_special_tokens=True)
+        final_result = self.__gt_post_processing(result, text)
+
+        return GenerationResult(text=final_result)
+
+
+
+    def __gt_post_processing(self, result, text):
```

Comment:
`post_process_generated_text` would be more descriptive

Reply:
667cfd4